### PR TITLE
Implement message signing rpc method

### LINKED ISSUE #23: Implement message signing rpc method
Method should receive message to sign, display confirmation prompt and return signature.

### PR DESCRIPTION
**Pull request by MakMuftic**
closes #23

### CODE REVIEW

File: packages/snap/test/unit/rpc/signMessage.test.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import chai, {expect} from "chai";
+import sinonChai from "sinon-chai";
+import {testAppKey} from "./keyPairTestConstants";
+import {SnapConfig} from "@nodefactory/metamask-filecoin-types";
+import {Message} from "@zondax/filecoin-signing-tools/js";
+import {signMessage} from "../../../src/rpc/signMessage";
+
+chai.use(sinonChai);
+
+describe('Test rpc handler function: getPublicKey', function () {
```
Comment (mpetrun5) [inline packages/snap/test/unit/rpc/signMessage.test.ts:11]: Shouldn't this be signMessage?

File: packages/snap/tsconfig.json [hunk 2]
```diff
@@ -15,7 +15,7 @@
       "@filecoin-shipyard/lotus-client-schema": [
         "./@types/lotus-client-schema"
       ],
-      "@zondax/filecoin-signing-tools": [
+      "@zondax/filecoin-signing-tools/js": [
```
Comment (mpetrun5) [inline packages/snap/tsconfig.json:18]: :smile:

File: packages/snap/test/unit/rpc/signMessage.test.ts (hunk 1 above)
Comment (MakMuftic) [reply packages/snap/test/unit/rpc/signMessage.test.ts:11]: Yap :+1:

**Comment by mpetrun5** [comment]
Shouldn't this have confirmation prompt like in polkadot metamask?

File: packages/snap/src/rpc/signMessage.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import {Message, SignedMessage, transactionSign} from "@zondax/filecoin-signing-tools/js";
+import {Wallet} from "../interfaces";
+import {getKeyPair} from "../filecoin/account";
+import {showConfirmationDialog} from "../util/confirmation";
+
+export async function signMessage(wallet: Wallet, message: Message): Promise<SignedMessage> {
+  const confirmation =
+      await showConfirmationDialog(wallet, "Do you want to sign message with your private key?");
```
Comment (MakMuftic) [inline packages/snap/src/rpc/signMessage.ts:8]: @mpetrun5 Not sure if this is the best message, what do you think? Maybe I can add some `message` details, but then I should add all properties and that seems to me as too much info on prompt.

File: packages/snap/test/unit/rpc/signMessage.test.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        const signedMessage = await signMessage(walletStub, unsignedMessage);
+        expect(walletStub.send).to.have.been.calledOnce;
+        expect(walletStub.getPluginState).to.have.been.calledOnce;
+        expect(walletStub.getAppKey).to.have.been.calledOnce;
+        expect(signedMessage.message).to.be.eq(unsignedMessage);
+        expect(signedMessage.signature.data).to.not.be.empty;
+        expect(signedMessage.signature.type).to.be.eq(1);
+    });
+
+    it('should successfully sign valid message on negative prompt', async function () {
```
Comment (mpetrun5) [inline packages/snap/test/unit/rpc/signMessage.test.ts:44]: Shouldn't :smile: :

File: packages/snap/src/rpc/signMessage.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import {Wallet} from "../interfaces";
+import {getKeyPair} from "../filecoin/account";
+import {showConfirmationDialog} from "../util/confirmation";
+
+export async function signMessage(wallet: Wallet, message: Message): Promise<SignedMessage> {
+  const keypair = await getKeyPair(wallet);
+  const tx = transactionSerialize(message);
+  const confirmation = await showConfirmationDialog(
+    wallet,
+    `Do you want to sign message ${tx} with account ${keypair.address}?`
```
Comment (MakMuftic) [inline packages/snap/src/rpc/signMessage.ts:11]: @mpetrun5 @mpetrunic is this better ?

Comment (mpetrunic) [reply packages/snap/src/rpc/signMessage.ts:11]: We should display formatted unserialized data. bunch of hex symbols won't help out wheter you are signing data you think you are signing :smile:

**Comment by MakMuftic** [comment]
@mpetrunic I agree with you, but I just think that we can show only a subset of message parameters in prompt (as prompt would become too big in my opinion). My suggestion is to show **from**, **to** and **value**, what do you think ?
```
   export interface Message {
        to: string;
        from: string;
        nonce: number;
        value: string;
        gasprice: string
        gaslimit: number;
        method: number;
        params?: any;
    }
```

**Comment by mpetrunic** [comment]
> @mpetrunic I agree with you, but I just think that we can show only a subset of message parameters in prompt (as prompt would become too big in my opinion). My suggestion is to show **from**, **to** and **value**, what do you think ?
> 
> ```
>    export interface Message {
>         to: string;
>         from: string;
>         nonce: number;
>         value: string;
>         gasprice: string
>         gaslimit: number;
>         method: number;
>         params?: any;
>     }
> ```

gasPrice also, as that is also param that can loose users funds

**Comment by MakMuftic** [comment]
@mpetrunic As I was trying different formating, this seems like the cleanest to me:
Screenshot from 2020-07-20 12-03-15

**Pull request merged**